completion/available/system.completion.bash

```diff
@@ -3,26 +3,43 @@
 # Loads the system's Bash completion modules.
 # If Homebrew is installed (OS X), its Bash completion modules are loaded.
 
-if [ -f /etc/bash_completion ]; then
-  . /etc/bash_completion
+if [ -r /etc/bash_completion ] ; then
+  # shellcheck disable=SC1091
+  source /etc/bash_completion
 fi
 
 # Some distribution makes use of a profile.d script to import completion.
-if [ -f /etc/profile.d/bash_completion.sh ]; then
-  . /etc/profile.d/bash_completion.sh
+if [ -r /etc/profile.d/bash_completion.sh ] ; then
+  # shellcheck disable=SC1091
+  source /etc/profile.d/bash_completion.sh
 fi
 
 
-if [ $(uname) = "Darwin" ] && command -v brew &>/dev/null ; then
-  BREW_PREFIX=$(brew --prefix)
+if [ "$(uname)" == "Darwin" ] ; then
 
-  if [ -f "$BREW_PREFIX"/etc/bash_completion ]; then
-    . "$BREW_PREFIX"/etc/bash_completion
+  # XCode installs git's completion files to a non-standard place
+  # Note: This should be loaded before brew's completion folder is parsed,
+  #       multiple tools glob on to this completion.
+  _git_comp='/Applications/Xcode.app/Contents/Developer/usr/share/git-core/git-completion.bash'
+  if _command_exists git && ! complete -p git &>/dev/null && [ -r "$_git_comp" ] ; then
+    # shellcheck disable=SC1090
+    source "$_git_comp"
   fi
+  unset _git_comp
 
- # homebrew/versions/bash-completion2 (required for projects.completion.bash) is installed to this path
-  if [ "${BASH_VERSINFO}" -ge 4 ] && [ -f "$BREW_PREFIX"/share/bash-completion/bash_completion ]; then
-    export BASH_COMPLETION_COMPAT_DIR="$BREW_PREFIX"/etc/bash_completion.d
-    . "$BREW_PREFIX"/share/bash-completion/bash_completion
+  if _command_exists brew ; then
+    BREW_PREFIX=$(brew --prefix)
+
+    if [ -r "$BREW_PREFIX"/etc/bash_completion ] ; then
+      # shellcheck disable=SC1090
+      source "$BREW_PREFIX"/etc/bash_completion
+    fi
+
+    # homebrew/versions/bash-completion2 (required for projects.completion.bash) is installed to this path
+    if [ -r "$BREW_PREFIX"/etc/profile.d/bash_completion.sh ] ; then
```

Comment:
For the case where you have both `$BREW_PREFIX/etc/bash_completion` and `$BREW_PREFIX/etc/profile.d/bash_completion.sh` (which I seem to have), it's loading both of them. Isn't that loading things twice in this case? Shouldn't they be mutually exclusive? Not sure I understand the difference between the two, I might be missing something.

Reply:
@nwinkler I thought at first too, but brew won't let you install both completions at once, so it's my understanding that it's one or the other, but never both. To that end, maybe this should be an `elif` instead?

Reply:
The interesting part is that I have both locations that are checked for here:

* I have stuff in `$BREW_PREFIX/etc/bash_completion`, which triggers the first `if`.
* I also have the additional script in `$BREW_PREFIX/etc/profile.d/bash_completion.sh`, which triggers the second `if`.

When I run `brew info bash-completion` locally, I get the following output:

```bash
bash-completion: stable 1.3 (bottled)
Programmable completion for Bash 3.2
https://salsa.debian.org/debian/bash-completion
Conflicts with:
  bash-completion@2 (because Differing version of same formula)
/usr/local/Cellar/bash-completion/1.3_3 (189 files, 608.2KB) *
  Poured from bottle on 2017-09-28 at 09:56:35
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/bash-completion.rb
==> Caveats
Add the following line to your ~/.bash_profile:
  [[ -r "/usr/local/etc/profile.d/bash_completion.sh" ]] && . "/usr/local/etc/profile.d/bash_completion.sh"

Bash completion has been installed to:
  /usr/local/etc/bash_completion.d
```

As you can see, it mentions the `$BREW_PREFIX/etc/profile.d/bash_completion.sh` script as well. It seems to use the same location as `bash-completion@2`. I'm wondering whether we need the first location `$BREW_PREFIX/etc/bash_completion` at all. Let me give that a try locally...

Reply:
Tried this locally just now. I think we can remove the first `if`, checking/sourcing the `$BREW_PREFIX/etc/bash_completion` script. The `$BREW_PREFIX/etc/profile.d/bash_completion.sh` script does exactly that already, it checks for the existence of the `etc/bash_completion` script and then runs that.

The way we're calling it at the moment (in this PR), basically loads everything twice. Let's get rid of the first `if` and load things the way they are recommended in the `brew` instructions. It's possible that things have changed since this code was originally added.

The same probably applies for the two `if` statements at the top of the file that check the same thing without `$BREW_PREFIX` - it should be either/or (or just the second one). I also wonder wether we could combine the two different branches (one with `$BREW_PREFIX`, the other one without) into one set of instructions - the only difference seems to be the `$BREW_PREFIX` variable at the start of the location.

Reply:
Yeah, this mirrors my morning's testing as well. However, I stumbled on one more edge case: My manually-installed versions, e.g. `helm@2` is installing bash completion a completely non-standard location. I'm going to try and inspect that a little more later, but have to pause for today.

Reply:
Good point. Doesn't change the fact that we can probably remove one of the duplicate locations.

The issue is that some tools (Xcode's `git`, `Docker.app`) install their completion files in non-standard locations. For these cases, we might want to have specific `completion` files, unless we want to handle all edge cases here (I don't think we want to do that...).

Reply:
Yeah I'm thinking the same thing. Probably better to leave them in their own files then.

Reply:
@cornfeedhobo I still think we should get rid of one of the `if` checks as described above. Let's keep the `$BREW_PREFIX/etc/profile.d/bash_completion.sh` and remove the `$BREW_PREFIX/etc/bash_completion` check/`source`...